src/settings/DatabaseHelper.ts

```diff
@@ -0,0 +1,134 @@
+import { Database } from 'sqlite3';
+
+export interface HasId extends Record<string, any> {
+  id: string;
+}
+
+export default abstract class DatabaseHelper<T extends HasId> {
+  #db: Database;
+
+  constructor(file: string) {
+    this.#db = new Database(file);
+  }
+
+  protected createTable(tableName: string, columns: string[]): Promise<void> {
+    let sql: string = `CREATE TABLE IF NOT EXISTS ${tableName} (`;
+    for (const [index, column] of columns.entries()) {
+      sql += column;
+      if (index !== columns.length - 1) sql += ', ';
+    }
+    sql += ')';
+    return this.runQuery(sql, []);
+  }
+
+  /**
+   * Inserts or updates an object in the table, depending on whether there is already a row with the id stored in the table
+   * @param tableName The table to insert or update to
+   * @param data The data to insert or update
+   */
+  protected async insertOrUpdate(tableName: string, data: T): Promise<void> {
```

Comment:
Implement [upsert](https://www.sqlite.org/lang_UPSERT.html) function in DatabaseHelper and use that here instead